avm/res/container-service/managed-cluster/agent-pool/main.bicep

```
@@ -151,11 +161,14 @@ param vnetSubnetResourceId string?
 @description('Optional. Determines the type of workload a node can run.')
 param workloadRuntime string?
 
-resource managedCluster 'Microsoft.ContainerService/managedClusters@2024-09-01' existing = {
+@description('Optional. Windows OS configuration.')
+param windowsProfile windowsProfileType?
```

Would recommend to use a ResourceDerivedType for this which is create for any parameter that doesn't need any customization:

```suggestion
param windowsProfile resourceInput<'Microsoft.ContainerService/managedClusters/agentPools@2025-05-02-preview'>.properties.windowsProfile?
```

Aside the added benefit of less maintenance, it also adds less to a template's size.

The same comment applies to the LinuxConfiguration.